File: javascript/test/Behavioral/TemplateMethod.js
(function() {
  define(['Behavioral/Template_Method/DvdTitleInfo', 'Behavioral/Template_Method/BookTitleInfo', 'Behavioral/Template_Method/GameTitleInfo'], function(DvdTitleInfo, BookTitleInfo, GameTitleInfo) {
    'use strict';
    return describe('Test Behavioral Template Method', function() {
      return it('Can create 3 title infos', function() {
        var bladeRunner, electricSheep, sheepRaider;
        bladeRunner = new DvdTitleInfo("Blade Runner", "Harrison Ford", '1');
        electricSheep = new BookTitleInfo("Do Androids Dream of Electric Sheep?", "Phillip K. Dick");
        sheepRaider = new GameTitleInfo("Sheep Raider");
        expect("Testing bladeRunner   " + bladeRunner.ProcessTitleInfo()).toEqual('Testing bladeRunner   DVD: Blade Runner, starring Harrison Ford ,, encoding region: 1');
        expect("Testing electricSheep " + electricSheep.ProcessTitleInfo()).toEqual('Testing electricSheep Book: Do Androids Dream of Electric Sheep?, Author: Phillip K. Dick, ');
        return expect("Testing sheepRaider   " + sheepRaider.ProcessTitleInfo()).toEqual('Testing sheepRaider   Game: Sheep Raider, ');
      });
    });
  });

}).call(this);

File: scala/src/test/scala/Behavioral/TemplateMethod.scala
//TestTitleInfoTemplate - testing the Template

package test.Behavioral;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Behavioral.Template_Method.*;



public class TemplateMethod {
   
	@Test 
   public void templatemethod() {
		TitleInfo bladeRunner = new DvdTitleInfo("Blade Runner", "Harrison Ford", '1'); 
		TitleInfo electricSheep = new BookTitleInfo("Do Androids Dream of Electric Sheep?", "Phillip K. Dick");        
		TitleInfo sheepRaider = new GameTitleInfo("Sheep Raider");

		org.junit.Assert.assertEquals("Testing bladeRunner" ,   
			"DVD: Blade Runner, starring Harrison Ford, encoding region: 1",
			bladeRunner.ProcessTitleInfo()
		);
		org.junit.Assert.assertEquals("Testing electricSheep" , 
			"Book: Do Androids Dream of Electric Sheep?, Author: Phillip K. Dick ",
			electricSheep.ProcessTitleInfo()
		);
		org.junit.Assert.assertEquals("Testing sheepRaider" ,
			"Game: Sheep Raider ",
			sheepRaider.ProcessTitleInfo()
		);
   }
}
